workers-site/index.js

```diff
@@ -58,6 +71,79 @@ async function handleEvent(event) {
   }
 }
 
+class StringInjector {
+  constructor(name, body) {
+    this.name = name
+    this.body = body
+  }
+
+  element(element){
+    element.prepend(`
+    <script>
+      ${ this.name } = \`${ this.escapeQuotes(this.body) }\`;
+    </script>
+    `, {
+      html: true
+    })
+  }
+
+  escapeQuotes(value) {
+    return value.replace(new RegExp('`', 'g'), '\\\`')
+  }
+}
+
+
+
+function template(rows){
+  function each(value, func){
+    let out = ''
+    for (let key in value){
+      out += func(value[key], key)
+    }
+    return out
+  }
+
+  function tbody(v){
+    return `
+      <tbody>
+        ${ each(v, row) }
+      </tbody>
+    `
+  }
+
+  function row(r){
+    return `
+      <tr data-status="${ r.status.replace(/ /g, '-') }">
+        ${ each(r, cell) }
+      </tr>
+    `
+  }
+
+  function cell(val, key){
+    return `
+      <td data-column="${ key }">${ val }
```

Comment:
I think you also have to set custom `data-value` attributes for some cols (`asn`, `status`).

Reply:
That’s correct to maintain parity with the existing generated markup. However, I’m realizing now that after I had added `data-sortable-type="numeric"` to the ASN column `<th/>`, the `data-value` attribute is now only needed for the status column.

Reply:
Cool, I believe the latest commit adds it.

Reply:
Unless I’m missing something, not quite. In order to sort the status column sensibly, we’re [converting "safe", "partially safe", and "unsafe" to integers, 1, 2, and 3, respectively](https://github.com/cloudflare/isbgpsafeyet.com/blob/a01d52ffaeb4c8db7a14b0760ab81e14ae734086/public/js/index.js#L25). We don’t need a `data-value` attribute for the other columns.

Reply:
Fixed